maven/lib/dependabot/maven/version.rb

```diff
@@ -175,8 +175,11 @@ def compare_token(token:, other_token:)
 
         return 1 if NAMED_QUALIFIERS_HIERARCHY[other_token]
 
-        token = token.to_i if token.match?(/^\d+$/)
-        other_token = other_token.to_i if other_token.match?(/^\d+$/)
+        if token.match?(/^\d+$/) && other_token.match?(/^\d+$/)
```

Comment:
Would it be safer to use `\A` and `\z` to match start and end of string to make sure the token doesn't have additional line endings injected in? It's not clear to me if token and/or other_token is considered user input or not.

WDYT?

Reply:
Yeah, seems safer 👍